lib/rbi/rewriters/filter_versions.rb

```diff
@@ -0,0 +1,112 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RBI
+  module Rewriters
+    # Take a gem version and filter out all RBI that is not relevant to that version based on @version annotations
```

Comment:
@Morriar Can you take a look at this comment and let me know what you think!

Reply:
The comment is really good 👍 

No need for an experimental mention, `rbi` just provides the rewriter, any tool that builds a feature on top of this may in the other hand mark it as experimental.